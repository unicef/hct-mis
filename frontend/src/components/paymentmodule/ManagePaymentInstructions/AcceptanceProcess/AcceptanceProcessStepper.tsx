import Step from '@material-ui/core/Step';
import StepLabel from '@material-ui/core/StepLabel';
import Stepper from '@material-ui/core/Stepper';
import CancelIcon from '@material-ui/icons/Cancel';
import React from 'react';
import { useTranslation } from 'react-i18next';
import styled from 'styled-components';

const StyledCancelIcon = styled(CancelIcon)`
  color: #e90202;
`;

interface AcceptanceProcessStepperProps {
  acceptanceProcess;
}

export const AcceptanceProcessStepper = ({
  acceptanceProcess,
}: AcceptanceProcessStepperProps): React.ReactElement => {
  const { t } = useTranslation();
  const {
    rejectedOn,
    actions,
    authorizationNumberRequired,
    releaseNumberRequired,
  } = acceptanceProcess;
  const steps = [
    {
      name: `${t('Authorization')} (${
        actions.authorization.length
      }/${authorizationNumberRequired})`,
      hasError: rejectedOn === 'IN_AUTHORIZATION',
      isCompleted: actions.authorization.length === authorizationNumberRequired,
    },
    {
      name: `${t('Release')} (${
        actions.release.length
      }/${releaseNumberRequired})`,
      hasError: rejectedOn === 'IN_REVIEW',
      isCompleted: actions.release.length === releaseNumberRequired,
    },
  ];
  const getActiveStep = (): number => {
    if (actions.release.length === releaseNumberRequired) {
      return 1;
    }
    return 0;
  };

  return (
    <Stepper activeStep={getActiveStep()}>
      {steps.map((step) => (
        <Step completed={step.isCompleted} key={step.name}>
          <StepLabel
            error={step.hasError}
            StepIconComponent={step.hasError ? StyledCancelIcon : null}
          >
            {step.name}
          </StepLabel>
        </Step>
      ))}
    </Stepper>
  );
};
