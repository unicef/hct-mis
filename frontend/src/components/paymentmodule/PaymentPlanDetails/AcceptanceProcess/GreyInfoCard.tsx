import { Box } from '@material-ui/core';
import React from 'react';
import { useTranslation } from 'react-i18next';
import styled from 'styled-components';

const GreyText = styled.div`
  color: #9e9e9e;
`;

const GreyBox = styled(Box)`
  background-color: #f4f5f6;
`;
interface GreyInfoCardProps {
  businessArea?: string;
  permissions?: string[];
}

export function GreyInfoCard({
  businessArea,
  permissions,
}: GreyInfoCardProps): React.ReactElement {
  const { t } = useTranslation();

  return (
    <GreyBox p={3}>
      Approved by someone<GreyText>on 01/01/2022</GreyText>
    </GreyBox>
  );
}
