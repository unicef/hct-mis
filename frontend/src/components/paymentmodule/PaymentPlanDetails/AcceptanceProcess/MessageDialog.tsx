import {
  Box,
  Button,
  DialogContent,
  DialogTitle,
  IconButton,
} from '@material-ui/core';
import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import MessageIcon from '@material-ui/icons/Message';
import styled from 'styled-components';
import { Dialog } from '../../../../containers/dialogs/Dialog';
import { DialogActions } from '../../../../containers/dialogs/DialogActions';

const DialogTitleWrapper = styled.div`
  border-bottom: 1px solid ${({ theme }) => theme.hctPalette.lighterGray};
`;

const DialogFooter = styled.div`
  padding: 12px 16px;
  margin: 0;
  border-top: 1px solid ${({ theme }) => theme.hctPalette.lighterGray};
  text-align: right;
`;

const GreyText = styled.div`
  color: #9e9e9e;
`;

const MessageIconContainer = styled(Box)`
  position: relative;
  top: 4px;
  font-size: 16px;
  color: #043f91;
`;

export interface MessageDialogProps {
  message: string;
  author: string;
  date: string;
}
export function MessageDialog({
  message,
  author,
  date,
}: MessageDialogProps): React.ReactElement {
  const { t } = useTranslation();
  const [MessageDialogOpen, setMessageDialogOpen] = useState(false);
  return (
    <>
      <Box p={2}>
        <IconButton size='small' onClick={() => setMessageDialogOpen(true)}>
          <MessageIconContainer>
            <MessageIcon fontSize='inherit' />
          </MessageIconContainer>
        </IconButton>
      </Box>
      <Dialog
        open={MessageDialogOpen}
        onClose={() => setMessageDialogOpen(false)}
        scroll='paper'
        aria-labelledby='form-dialog-title'
        maxWidth='md'
      >
        <DialogTitleWrapper>
          <DialogTitle id='scroll-dialog-title'>{t('Comment')}</DialogTitle>
        </DialogTitleWrapper>
        <DialogContent>
          <Box display='flex' flexDirection='column'>
            <Box display='flex'>
              {author}{' '}
              <GreyText>
                <Box ml={1}>on {date}</Box>
              </GreyText>
            </Box>
            {message}
          </Box>
        </DialogContent>
        <DialogFooter>
          <DialogActions>
            <Button
              type='submit'
              color='primary'
              variant='contained'
              onClick={() => setMessageDialogOpen(false)}
            >
              {t('Close')}
            </Button>
          </DialogActions>
        </DialogFooter>
      </Dialog>
    </>
  );
}
