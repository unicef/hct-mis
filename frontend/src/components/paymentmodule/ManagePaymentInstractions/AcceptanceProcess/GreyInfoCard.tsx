import { Box } from '@material-ui/core';
import React from 'react';
import styled from 'styled-components';
import { UniversalMoment } from '../../../core/UniversalMoment';
import { ApprovalNode } from '../../../../__generated__/graphql';
import { MessageDialog } from './MessageDialog';

const GreyText = styled.div`
  color: #9e9e9e;
`;

const GreyTitle = styled.div`
  color: #7c8990;
  text-transform: uppercase;
  font-size: 12px;
`;

const IconPlaceholder = styled.div`
  width: 16px;
  height: 16px;
`;

const GreyBox = styled(Box)`
  background-color: #f4f5f6;
`;

interface GreyInfoCardProps {
  topMessage: string;
  topDate: string;
  approvals: ApprovalNode[];
}

export const GreyInfoCard = ({
  topMessage,
  topDate,
  approvals,
}: GreyInfoCardProps): React.ReactElement => {
  const mappedApprovals = approvals
    .filter((action) => action.info)
    .map((action) => {
      const { info, createdAt, comment, createdBy } = action;
      return (
        <Box alignItems='center' display='flex' key={createdAt}>
          {info}
          <Box ml={1}>
            <GreyText>
              on <UniversalMoment>{createdAt}</UniversalMoment>
            </GreyText>
          </Box>
          <Box p={1} ml={1}>
            {comment ? (
              <MessageDialog
                comment={comment}
                author={createdBy}
                date={createdAt}
              />
            ) : (
              <IconPlaceholder />
            )}
          </Box>
        </Box>
      );
    });

  return (
    <Box display='flex' flexDirection='column'>
      <Box p={3}>
        <GreyTitle>
          {topMessage} on <UniversalMoment>{topDate}</UniversalMoment>
        </GreyTitle>
      </Box>
      <GreyBox
        display='flex'
        flexDirection='column'
        alignItems='center'
        ml={3}
        mr={3}
        p={3}
      >
        {mappedApprovals}
      </GreyBox>
    </Box>
  );
};
