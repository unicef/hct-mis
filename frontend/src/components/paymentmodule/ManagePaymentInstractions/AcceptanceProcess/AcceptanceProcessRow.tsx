import { Box, Grid } from '@material-ui/core';
import React from 'react';
import { useTranslation } from 'react-i18next';
import { renderUserName } from '../../../../utils/utils';
import { DividerLine } from '../../../core/DividerLine';
import { AcceptanceProcessStepper } from './AcceptanceProcessStepper';
import { GreyInfoCard } from './GreyInfoCard';

interface AcceptanceProcessRowProps {
  acceptanceProcess;
  showDivider?: boolean;
}

export const AcceptanceProcessRow = ({
  acceptanceProcess,
  showDivider = false,
}: AcceptanceProcessRowProps): React.ReactElement => {
  const { t } = useTranslation();
  const {
    actions,
    sentForAuthorizationDate,
    sentForAuthorizationBy,
    sentForReleaseDate,
    sentForReleaseBy,
    rejectedOn,
  } = acceptanceProcess;

  return (
    <Box>
      <AcceptanceProcessStepper acceptanceProcess={acceptanceProcess} />
      <Grid container>
        <Grid item xs={6}>
          {actions.authorization.length > 0 && (
            <GreyInfoCard
              topMessage={`Sent for authorization by ${renderUserName(
                sentForAuthorizationBy,
              )}`}
              topDate={sentForAuthorizationDate}
              approvals={actions.authorization}
            />
          )}
        </Grid>
        <Grid item xs={6}>
          {actions.release.length > 0 && (
            <GreyInfoCard
              topMessage={`Sent for review by ${renderUserName(
                sentForReleaseBy,
              )}`}
              topDate={sentForReleaseDate}
              approvals={actions.release}
            />
          )}
        </Grid>
        {actions.reject.length > 0 && (
          <Grid container>
            <Grid item xs={4}>
              {rejectedOn === 'IN_AUTHORIZATION' && (
                <GreyInfoCard
                  topMessage={t('Rejected in Authorization stage')}
                  topDate={actions.reject[0]?.createdAt}
                  approvals={actions.reject}
                />
              )}
            </Grid>
            <Grid item xs={4}>
              {rejectedOn === 'IN_REVIEW' && (
                <GreyInfoCard
                  topMessage={t('Rejected in Release stage')}
                  topDate={actions.reject[0]?.createdAt}
                  approvals={actions.reject}
                />
              )}
            </Grid>
          </Grid>
        )}
      </Grid>
      {showDivider && <DividerLine />}
    </Box>
  );
};
