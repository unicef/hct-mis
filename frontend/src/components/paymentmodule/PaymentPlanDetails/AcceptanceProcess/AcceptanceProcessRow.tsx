import { Box, Grid } from '@material-ui/core';
import React from 'react';
import { useTranslation } from 'react-i18next';
import styled from 'styled-components';
import { renderUserName } from '../../../../utils/utils';
import { PaymentPlanQuery } from '../../../../__generated__/graphql';
import { DividerLine } from '../../../core/DividerLine';
import { AcceptanceProcessStepper } from './AcceptanceProcessStepper/AcceptanceProcessStepper';
import { GreyInfoCard } from './GreyInfoCard';

const StyledBox = styled(Box)`
  width: 100%;
`;

interface AcceptanceProcessRowProps {
  acceptanceProcess: PaymentPlanQuery['paymentPlan']['approvalProcess']['edges'][0]['node'];
  paymentPlan: PaymentPlanQuery['paymentPlan'];
}

export const AcceptanceProcessRow = ({
  acceptanceProcess,
  paymentPlan,
}: AcceptanceProcessRowProps): React.ReactElement => {
  const { t } = useTranslation();
  const {
    actions,
    sentForApprovalDate,
    sentForApprovalBy,
    sentForAuthorizationDate,
    sentForAuthorizationBy,
    sentForFinanceReviewDate,
    sentForFinanceReviewBy,
    rejectedOn,
  } = acceptanceProcess;

  const {
    approvalNumberRequired,
    authorizationNumberRequired,
    financeReviewNumberRequired,
  } = paymentPlan;

  const getRejectedOnString = (stage: string): string => {
    switch (stage) {
      case 'IN_APPROVAL':
        return t('Rejected in Approval stage');
      case 'IN_AUTHORIZATION':
        return t('Rejected in Authorization stage');
      case 'IN_REVIEW':
        return t('Rejected in Finance Review stage');

      default:
        return '';
    }
  };

  return (
    <StyledBox m={5}>
      <AcceptanceProcessStepper
        approvalNumberRequired={approvalNumberRequired}
        authorizationNumberRequired={authorizationNumberRequired}
        financeReviewNumberRequired={financeReviewNumberRequired}
        acceptanceProcess={acceptanceProcess}
      />
      <Grid container>
        <Grid item xs={4}>
          {actions.approval.length > 0 && (
            <GreyInfoCard
              topMessage={`Sent for approval by ${renderUserName(
                sentForApprovalBy,
              )}`}
              topDate={sentForApprovalDate}
              approvals={actions.approval}
            />
          )}
        </Grid>
        <Grid item xs={4}>
          {actions.authorization.length > 0 && (
            <GreyInfoCard
              topMessage={`Sent for authorization by ${renderUserName(
                sentForAuthorizationBy,
              )}`}
              topDate={sentForAuthorizationDate}
              approvals={actions.authorization}
            />
          )}
        </Grid>
        <Grid item xs={4}>
          {actions.financeReview.length > 0 && (
            <GreyInfoCard
              topMessage={`Sent for review by ${renderUserName(
                sentForFinanceReviewBy,
              )}`}
              topDate={sentForFinanceReviewDate}
              approvals={actions.financeReview}
            />
          )}
        </Grid>
        {actions.reject.length > 0 && (
          <Grid container>
            <Grid item xs={4}>
              {rejectedOn === 'IN_APPROVAL' && (
                <GreyInfoCard
                  topMessage={getRejectedOnString(rejectedOn)}
                  topDate={actions.reject[0]?.createdAt}
                  approvals={actions.reject}
                />
              )}
            </Grid>
            <Grid item xs={4}>
              {rejectedOn === 'IN_AUTHORIZATION' && (
                <GreyInfoCard
                  topMessage={getRejectedOnString(rejectedOn)}
                  topDate={actions.reject[0]?.createdAt}
                  approvals={actions.reject}
                />
              )}
            </Grid>
            <Grid item xs={4}>
              {rejectedOn === 'IN_REVIEW' && (
                <GreyInfoCard
                  topMessage={getRejectedOnString(rejectedOn)}
                  topDate={actions.reject[0]?.createdAt}
                  approvals={actions.reject}
                />
              )}
            </Grid>
          </Grid>
        )}
      </Grid>
      <DividerLine />
    </StyledBox>
  );
};
