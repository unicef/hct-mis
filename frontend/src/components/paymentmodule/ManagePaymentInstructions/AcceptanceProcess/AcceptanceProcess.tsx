import { Box, Button, Typography } from '@material-ui/core';
import ExpandLessIcon from '@material-ui/icons/ExpandLess';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import styled from 'styled-components';
import { Title } from '../../../core/Title';
import { AcceptanceProcessRow } from './AcceptanceProcessRow';

const ButtonContainer = styled(Box)`
  width: 200px;
`;

interface AcceptanceProcessProps {
  records;
}

export const AcceptanceProcess = ({
  records,
}: AcceptanceProcessProps): React.ReactElement => {
  const { t } = useTranslation();
  const [showAll, setShowAll] = useState(false);

  const renderRows = (): React.ReactElement => {
    if (showAll) {
      return records.map(({ node }) => (
        <AcceptanceProcessRow
          key={node.id}
          acceptanceProcess={node}
          showDivider
        />
      ));
    }
    return <AcceptanceProcessRow acceptanceProcess={records[0].node} />;
  };

  const renderShowMore = (): React.ReactElement | boolean => {
    if (records.length <= 1) {
      return false;
    }

    if (showAll) {
      return (
        <ButtonContainer>
          <Button
            variant='outlined'
            color='primary'
            onClick={() => setShowAll(!showAll)}
            endIcon={<ExpandLessIcon />}
            data-cy='btn-hide'
          >
            {t('HIDE')}
          </Button>
        </ButtonContainer>
      );
    }

    return (
      <ButtonContainer>
        <Button
          variant='outlined'
          color='primary'
          onClick={() => setShowAll(!showAll)}
          endIcon={<ExpandMoreIcon />}
          data-cy='btn-show-previous'
        >
          {t('SHOW PREVIOUS')}
        </Button>
      </ButtonContainer>
    );
  };

  if (!records.length) {
    return null;
  }

  return (
    <Box m={5}>
      <Box display='flex' justifyContent='space-between' mt={4}>
        <Title>
          <Typography variant='h6' data-cy='title-acceptance-process'>
            {t('Acceptance Process')}
          </Typography>
        </Title>
      </Box>
      {renderRows()}
      {renderShowMore()}
    </Box>
  );
};
