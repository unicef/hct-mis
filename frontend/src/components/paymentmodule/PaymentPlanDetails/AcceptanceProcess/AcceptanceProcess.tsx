import { Box, Button, Typography } from '@material-ui/core';
import styled from 'styled-components';
import ExpandLessIcon from '@material-ui/icons/ExpandLess';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { PaymentPlanQuery } from '../../../../__generated__/graphql';
import { ContainerColumnWithBorder } from '../../../core/ContainerColumnWithBorder';
import { Title } from '../../../core/Title';
import { AcceptanceProcessRow } from './AcceptanceProcessRow';

const ButtonContainer = styled(Box)`
  width: 200px;
`;

interface AcceptanceProcessProps {
  paymentPlan: PaymentPlanQuery['paymentPlan'];
}

export const AcceptanceProcess = ({
  paymentPlan,
}: AcceptanceProcessProps): React.ReactElement => {
  const { t } = useTranslation();
  const { edges } = paymentPlan.approvalProcess;
  const [showAll, setShowAll] = useState(false);

  const matchDataSize = (data: PaymentPlanQuery['paymentPlan']['approvalProcess']['edges']): PaymentPlanQuery['paymentPlan']['approvalProcess']['edges'] => {
    return showAll ? data : [data[0]];
  };

  if (!edges.length) {
    return null;
  }

  return (
    <Box m={5}>
      <ContainerColumnWithBorder>
        <Box mt={4}>
          <Title>
            <Typography variant='h6'>{t('Acceptance Process')}</Typography>
          </Title>
        </Box>
        {matchDataSize(edges).map((edge) => (
          <AcceptanceProcessRow
            key={edge.node.id}
            acceptanceProcess={edge.node}
            paymentPlan={paymentPlan}
          />
        ))}
        {edges.length > 1 && (
          <ButtonContainer>
            <Button
              variant='outlined'
              color='primary'
              onClick={() => setShowAll(!showAll)}
              endIcon={showAll ? <ExpandLessIcon /> : <ExpandMoreIcon />}
            >
              {showAll ? t('HIDE') : t('SHOW PREVIOUS')}
            </Button>
          </ButtonContainer>
        )}
      </ContainerColumnWithBorder>
    </Box>
  );
};
