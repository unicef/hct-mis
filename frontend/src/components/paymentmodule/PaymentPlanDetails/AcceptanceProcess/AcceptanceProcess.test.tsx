import React from 'react';
import { fakeApolloPaymentPlan } from '../../../../../fixtures/payments/fakeApolloPaymentPlan';
import { render } from '../../../../testUtils/testUtils';
import { AcceptanceProcess } from './AcceptanceProcess';

describe('components/paymentmodule/PaymentPlanDetails/AcceptanceProcess', () => {
  it('should render', () => {
    const { container } = render(
      <AcceptanceProcess paymentPlan={fakeApolloPaymentPlan} />,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render empty without Approval Process', () => {
    const { container } = render(
      <AcceptanceProcess paymentPlan={{...fakeApolloPaymentPlan, approvalProcess: {...fakeApolloPaymentPlan.approvalProcess, edges: []}}} />,
    );
    expect(container).toMatchSnapshot();
  });
});
