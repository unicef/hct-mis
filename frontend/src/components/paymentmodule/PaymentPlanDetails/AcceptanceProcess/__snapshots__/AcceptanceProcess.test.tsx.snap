// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/paymentmodule/PaymentPlanDetails/AcceptanceProcess should render 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-1"
  >
    <div
      class="MuiPaper-root sc-bdVaJa fUBeDk MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiBox-root MuiBox-root-2"
      >
        <div
          class="sc-bwzfXH glegqy"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6"
          >
            Acceptance Process
          </h6>
        </div>
      </div>
      <div
        class="MuiBox-root MuiBox-root-3 sc-jTzLTM hDLiwJ"
      >
        <div
          class="MuiPaper-root MuiStepper-root MuiStepper-horizontal MuiPaper-elevation0"
        >
          <div
            class="MuiStep-root MuiStep-horizontal MuiStep-completed"
          >
            <span
              class="MuiStepLabel-root MuiStepLabel-horizontal"
            >
              <span
                class="MuiStepLabel-iconContainer"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiStepIcon-root MuiStepIcon-completed"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 0a12 12 0 1 0 0 24 12 12 0 0 0 0-24zm-2 17l-5-5 1.4-1.4 3.6 3.6 7.6-7.6L19 8l-9 9z"
                  />
                </svg>
              </span>
              <span
                class="MuiStepLabel-labelContainer"
              >
                <span
                  class="MuiTypography-root MuiStepLabel-label MuiStepLabel-completed MuiTypography-body2 MuiTypography-displayBlock"
                >
                  Approval (1/1)
                </span>
              </span>
            </span>
          </div>
          <div
            class="MuiStepConnector-root MuiStepConnector-horizontal MuiStepConnector-active"
          >
            <span
              class="MuiStepConnector-line MuiStepConnector-lineHorizontal"
            />
          </div>
          <div
            class="MuiStep-root MuiStep-horizontal"
          >
            <span
              class="MuiStepLabel-root MuiStepLabel-horizontal"
            >
              <span
                class="MuiStepLabel-iconContainer"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiStepIcon-root MuiStepIcon-active"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="12"
                  />
                  <text
                    class="MuiStepIcon-text"
                    text-anchor="middle"
                    x="12"
                    y="16"
                  >
                    2
                  </text>
                </svg>
              </span>
              <span
                class="MuiStepLabel-labelContainer"
              >
                <span
                  class="MuiTypography-root MuiStepLabel-label MuiStepLabel-active MuiTypography-body2 MuiTypography-displayBlock"
                >
                  Authorization (0/1)
                </span>
              </span>
            </span>
          </div>
          <div
            class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled"
          >
            <span
              class="MuiStepConnector-line MuiStepConnector-lineHorizontal"
            />
          </div>
          <div
            class="MuiStep-root MuiStep-horizontal"
          >
            <span
              class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled"
            >
              <span
                class="MuiStepLabel-iconContainer"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiStepIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="12"
                  />
                  <text
                    class="MuiStepIcon-text"
                    text-anchor="middle"
                    x="12"
                    y="16"
                  >
                    3
                  </text>
                </svg>
              </span>
              <span
                class="MuiStepLabel-labelContainer"
              >
                <span
                  class="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
                >
                  Finance Release (0/1)
                </span>
              </span>
            </span>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
          >
            <div
              class="MuiBox-root MuiBox-root-4"
            >
              <div
                class="MuiBox-root MuiBox-root-5"
              >
                <div
                  class="sc-gZMcBi iLflOl"
                >
                  Sent for approval by Root Rootkowski
                   on 
                  <time
                    datetime="1661843731428"
                  >
                    30 Aug 2022
                  </time>
                </div>
              </div>
              <div
                class="MuiBox-root MuiBox-root-6 sc-VigVT dYssCQ"
              >
                <div
                  class="MuiBox-root MuiBox-root-7"
                >
                  Approved by Root Rootkowski
                  <div
                    class="MuiBox-root MuiBox-root-8"
                  >
                    <div
                      class="sc-iwsKbI cmTGgT"
                    >
                      on 
                      <time
                        datetime="1661843741787"
                      >
                        30 Aug 2022
                      </time>
                    </div>
                  </div>
                  <div
                    class="MuiBox-root MuiBox-root-9"
                  >
                    <div
                      class="sc-gqjmRU dAXogI"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
          />
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`components/paymentmodule/PaymentPlanDetails/AcceptanceProcess should render empty without Approval Process 1`] = `<div />`;
